### PR TITLE
DDCNL-3747: Bootstrap and Play upgrade

### PR DESCRIPTION
**Pull request by Christian-QA**

### CODE REVIEW

File: test/services/UpdateAddressResponseSpec.scala [hunk 1]
```diff
@@ -46,27 +49,27 @@ class UpdateAddressResponseSpec extends BaseSpec with I18nSupport with MockitoSu
   def genericFunc(): Result =
     Ok
 
-  "UpdateAddressResponse.response" should {
+  "UpdateAddressResponse.response" must {
     "return the block result for UpdateAddressSuccessResponse" in {
-      val result = UpdateAddressSuccessResponse.response(genericFunc)
-      status(result) shouldBe OK
+      val result = Future(UpdateAddressSuccessResponse.response(genericErrors, genericFunc))
```
Comment (struong) [inline test/services/UpdateAddressResponseSpec.scala:54]: Why has Future been wrapped around this?

File: test/services/UpdateAddressResponseSpec.scala [hunk 2]
```diff
@@ -46,27 +49,27 @@ class UpdateAddressResponseSpec extends BaseSpec with I18nSupport with MockitoSu
...
+      status(result) mustBe OK
     }
 
     "return BAD_REQUEST for UpdateAddressBadRequestResponse" in {
-      val result = UpdateAddressBadRequestResponse.response(genericFunc)
-      status(result) shouldBe BAD_REQUEST
+      val result = Future(UpdateAddressBadRequestResponse.response(genericErrors, genericFunc))
```
Comment (struong) [inline test/services/UpdateAddressResponseSpec.scala:59]: Why has Future been wrapped around this?

File: test/services/UpdateAddressResponseSpec.scala [hunk 3]
```diff
@@ -46,27 +49,27 @@ class UpdateAddressResponseSpec extends BaseSpec with I18nSupport with MockitoSu
...
+      status(result) mustBe BAD_REQUEST
     }
 
     "return INTERNAL_SERVER_ERROR for UpdateAddressUnexpectedResponse" in {
       val updateAddressResponse = UpdateAddressUnexpectedResponse(HttpResponse(123))
-      val result = updateAddressResponse.response(genericFunc)
-      status(result) shouldBe INTERNAL_SERVER_ERROR
+      val result = Future(updateAddressResponse.response(genericErrors, genericFunc))
```
Comment (struong) [inline test/services/UpdateAddressResponseSpec.scala:65]: Why has Future been wrapped around this?

File: test/services/UpdateAddressResponseSpec.scala [hunk 4]
```diff
@@ -46,27 +49,27 @@ class UpdateAddressResponseSpec extends BaseSpec with I18nSupport with MockitoSu
...
+      status(result) mustBe INTERNAL_SERVER_ERROR
     }
 
     "return INTERNAL_SERVER_ERROR for UpdateAddressErrorResponse" in {
       val updateAddressResponse = UpdateAddressErrorResponse(new RuntimeException("not used"))
-      val result = updateAddressResponse.response(genericFunc)
-      status(result) shouldBe INTERNAL_SERVER_ERROR
+      val result = Future(updateAddressResponse.response(genericErrors, genericFunc))
```
Comment (struong) [inline test/services/UpdateAddressResponseSpec.scala:71]: Why has Future been wrapped around this?

File: test/services/http/SimpleHttpSpec.scala [hunk 5]
```diff
@@ -16,29 +16,25 @@
...
+import org.mockito.Mockito.{times, verify, when}
+import org.scalatestplus.mockito.MockitoSugar.mock
 import play.api.libs.json.Writes
 import uk.gov.hmrc.http.{BadGatewayException, HeaderCarrier, HttpResponse}
 import uk.gov.hmrc.play.bootstrap.http.DefaultHttpClient
 import util.BaseSpec
 
-import scala.concurrent.Future
-import scala.concurrent.ExecutionContext.Implicits.global
+import java.util.concurrent.LinkedBlockingQueue
```
Comment (struong) [inline test/services/http/SimpleHttpSpec.scala:28]: unused?

File: test/util/Fixtures.scala [hunk 6]
```diff
@@ -300,41 +299,44 @@ object Fixtures extends PafFixtures with TaiFixtures with CitizenDetailsFixtures
...
+  when(mockPartialRetriever.getPartialContent(any(), any(), any())(any(), any())) thenReturn Html("")
+
   val configValues =
     Map(
       "cookie.encryption.key"         -> "gvBoGdgzqG1AarzF1LY0zQ==",
       "sso.encryption.key"            -> "gvBoGdgzqG1AarzF1LY0zQ==",
       "queryParameter.encryption.key" -> "gvBoGdgzqG1AarzF1LY0zQ==",
       "json.encryption.key"           -> "gvBoGdgzqG1AarzF1LY0zQ==",
-      "metrics.enabled"               -> false
+      "metrics.enabled"               -> "false"
```
Comment (struong) [inline test/util/Fixtures.scala:319]: Why was this changed?

File: test/config/ConfigDecoratorSpec.scala [hunk 7]
```diff
@@ -17,34 +17,34 @@
 package config
 
 import java.net.{MalformedURLException, URL}
-
-import play.api.i18n.Langs
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
```
Comment (Christian-QA) [inline test/config/ConfigDecoratorSpec.scala:20]: Unused, was moved to BaseSpec

File: test/util/MessagesSpec.scala [hunk 8]
```diff
@@ -17,31 +17,30 @@
...
 import play.api.i18n.MessagesApi
-import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
 
 import scala.util.matching.Regex
 
-class MessagesSpec extends UnitSpec with WithFakeApplication {
+class MessagesSpec extends BaseSpec {
 
-  lazy val messagesAPI = fakeApplication.injector.instanceOf[MessagesApi]
+  lazy val messagesAPI = app.injector.instanceOf[MessagesApi]
```
Comment (Christian-QA) [inline test/util/MessagesSpec.scala:25]: Would it be worth changing instances of app.injector.instandOf to injected given both are used throughout the project and do the same thing?

File: test/views/html/integration/mainContentHeaderSpec.scala [hunk 9]
```diff
@@ -18,87 +18,67 @@ package views.html.integration
 
 import config.ConfigDecorator
 import org.joda.time.DateTime
-import org.scalatestplus.mockito.MockitoSugar
-import util.BaseSpec
 import views.html.ViewSpec
 
-class mainContentHeaderSpec extends ViewSpec with MockitoSugar {
+class mainContentHeaderSpec extends ViewSpec {
```
Comment (Christian-QA) [inline test/views/html/integration/mainContentHeaderSpec.scala:23]: Wrong casing

File: app/views/UnauthenticatedMainView.scala.html [hunk 10]
```diff
@@ -14,17 +14,16 @@
  * limitations under the License.
```
Comment (Christian-QA) [inline app/views/UnauthenticatedMainView.scala.html:14]: Wrong class casing

File: app/config/ConfigDecorator.scala [hunk 11]
```diff
@@ -245,9 +245,12 @@ class ConfigDecorator @Inject()(
...
     .get[String](s"assets.version") + '/'
 
-  lazy val sessionTimeoutInSeconds = runModeConfiguration.getOptional[Int]("session.timeout").getOrElse(900)
+  lazy val sessionTimeoutInSeconds = runModeConfiguration.getOptional[Int]("ptaSession.timeout").getOrElse(900)
   lazy val sessionTimeoutInMinutes = sessionTimeoutInSeconds / 60
-  lazy val sessionCountdownInSeconds = runModeConfiguration.getOptional[Int]("session.countdown").getOrElse(120)
+  lazy val sessionCountdownInSeconds = runModeConfiguration.getOptional[Int]("ptaSession.countdown").getOrElse(120)
+
+  lazy val serviceManagerRunModeFlag =
+    runModeConfiguration.getOptional[Boolean]("safeRedirectUrl.allowAbsolute").getOrElse(false)
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/config/ConfigDecorator.scala:253]: Done

File: app/controllers/address/StartDateController.scala [hunk 12]
```diff
@@ -86,8 +86,7 @@ class StartDateController @Inject()(
                   personDetails.address match {
                     case Some(Address(_, _, _, _, _, _, _, Some(currentStartDate), _, _)) =>
                       if (!currentStartDate.isBefore(proposedStartDate)) {
-                        BadRequest(
-                          cannotUpdateAddressView(typ, LanguageHelper.langUtils.Dates.formatDate(proposedStartDate)))
+                        BadRequest(cannotUpdateAddressView(typ, languageUtils.Dates.formatDate(proposedStartDate)))
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/controllers/address/StartDateController.scala:89]: Done

File: app/views/personaldetails/ReviewChangesView.scala.html [hunk 13]
```diff
@@ -87,7 +89,7 @@ <h1 class="heading-xlarge">@messages("label.check_your_details")</h1>
                 <dt class="cya-question">
                   @typ.ifIs(primary = messages("label.when_this_became_your_main_home"), sole = messages("label.when_you_started_living_here"), postal = "")
                 </dt>
-                <dd class="cya-answer">@LanguageHelper.langUtils.Dates.formatDate(startDateDto.startDate)</dd>
+                <dd class="cya-answer">@languageUtils.Dates.formatDate(startDateDto.startDate)</dd>
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/personaldetails/ReviewChangesView.scala.html:92]: Done

File: app/services/UpdateAddressResponse.scala [hunk 14]
```diff
@@ -22,18 +22,16 @@ import error.GenericErrors
...
     configDecorator: ConfigDecorator,
-    partialRetriever: LocalPartialRetriever,
     messages: Messages): Result =
     this match {
-      case UpdateAddressBadRequestResponse    => GenericErrors.badRequest
-      case UpdateAddressUnexpectedResponse(_) => GenericErrors.internalServerError
-      case UpdateAddressErrorResponse(_)      => GenericErrors.internalServerError
+      case UpdateAddressBadRequestResponse    => genericErrors.badRequest
+      case UpdateAddressUnexpectedResponse(_) => genericErrors.internalServerError
+      case UpdateAddressErrorResponse(_)      => genericErrors.internalServerError
```
Comment (brendanstraughan) [inline app/services/UpdateAddressResponse.scala:34]: Since these 2 cases do the same thing we can merge them together like this
case UpdateAddressUnexpectedResponse(_) || UpdateAddressErrorResponse(_) => genericErrors.internalServerError

File: app/config/ConfigDecorator.scala [hunk 15]
```diff
@@ -245,9 +245,12 @@ class ConfigDecorator @Inject()(
   lazy val assetsPrefix = runModeConfiguration.get[String](s"assets.url") + runModeConfiguration
     .get[String](s"assets.version") + '/'
 
-  lazy val sessionTimeoutInSeconds = runModeConfiguration.getOptional[Int]("session.timeout").getOrElse(900)
+  lazy val sessionTimeoutInSeconds = runModeConfiguration.getOptional[Int]("ptaSession.timeout").getOrElse(900)
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/config/ConfigDecorator.scala:248]: Done

File: app/controllers/ApplicationController.scala [hunk 16]
```diff
@@ -98,26 +99,31 @@ class ApplicationController @Inject()(
...
           Future.successful(BadRequest(technicalIssuesView(retryUrl)))
       }
     }
 
   private def logErrorMessage(reason: String): Unit =
-    Logger.warn(s"Unable to confirm user identity: $reason")
+    logger.warn(s"Unable to confirm user identity: $reason")
 
-  def signout(continueUrl: Option[SafeRedirectUrl], origin: Option[Origin]): Action[AnyContent] =
+  def signout(continueUrl: Option[RedirectUrl], origin: Option[Origin]): Action[AnyContent] =
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/controllers/ApplicationController.scala:118]: Done

File: app/services/partials/FormPartialService.scala [hunk 17]
```diff
@@ -33,9 +32,9 @@ class FormPartialService @Inject()(
   override val http: HttpClient,
   val metrics: Metrics,
   val configDecorator: ConfigDecorator,
-  sessionCookieCrypto: SessionCookieCrypto
+  headerCarrierForPartialsConverter: HeaderCarrierForPartialsConverter,
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/services/partials/FormPartialService.scala:35]: Done

File: app/util/AuditServiceTools.scala [hunk 18]
```diff
@@ -51,7 +51,9 @@ object AuditServiceTools {
     DataEvent(
       auditSource = auditSource,
       auditType = auditType,
-      tags = hc.headers.toMap ++ customTags.map(x => x._2.map((x._1, _))).flatten.toMap,
+      tags = hc
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/util/AuditServiceTools.scala:54]: Done

File: app/views/cards/home/TaxCalculationView.scala.html [hunk 19]
```diff
@@ -17,27 +17,29 @@
...
-@this()
+@import views.util.ViewUtils
+
+@this(viewUtils: ViewUtils)
 @(viewModel: TaxCalculationViewModel)(implicit messages: play.api.i18n.Messages, configDecorator: ConfigDecorator)
 
     @card(
         id = Some(s"taxcalc-${viewModel.taxYears.previousTaxYear}-${viewModel.taxYears.currentTaxYear}-card"),
-        url = Some(fromUrl(viewModel.heading.url)),
+        url = Some(viewUtils.fromUrl(viewModel.heading.url)),
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/cards/home/TaxCalculationView.scala.html:27]: Done

File: app/views/iv/failure/TimeOutView.scala.html [hunk 20]
```diff
@@ -16,13 +16,14 @@
...
+@import scala.concurrent.ExecutionContext
 
-@this()
+@this(ivOutcomeWrapperView: IvOutcomeWrapperView)
 
-@(retryUrl: String)(implicit request: Request[_], configDecorator: ConfigDecorator, partialRetriever: LocalPartialRetriever, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages)
+@(retryUrl: String)(implicit request: Request[_], configDecorator: ConfigDecorator, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages, ec: ExecutionContext)
 
-@views.html.iv.ivOutcomeWrapper(messages("label.you_have_been_signed_out"), messages("label.you_have_been_signed_out")) {
+@ivOutcomeWrapperView(messages("label.you_have_been_signed_out"), messages("label.you_have_been_signed_out")) {
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/iv/failure/TimeOutView.scala.html:26]: Done

File: app/views/iv/failure/TechnicalIssuesView.scala.html [hunk 21]
```diff
@@ -16,13 +16,14 @@
...
 @import uk.gov.hmrc.renderer.TemplateRenderer
-@import util.LocalPartialRetriever
+@import views.html.iv.IvOutcomeWrapperView
+@import scala.concurrent.ExecutionContext
 
-@this()
+@this(ivOutcomeWrapperView: IvOutcomeWrapperView)
 
-@(retryUrl: String)(implicit request: Request[_], configDecorator: ConfigDecorator, partialRetriever: LocalPartialRetriever, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages)
+@(retryUrl: String)(implicit request: Request[_], configDecorator: ConfigDecorator, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages, ec: ExecutionContext)
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/iv/failure/TechnicalIssuesView.scala.html:24]: Done

File: app/views/iv/failure/LockedOutView.scala.html [hunk 22]
```diff
@@ -16,13 +16,14 @@
...
+@import scala.concurrent.ExecutionContext
 
-@this()
+@this(ivOutcomeWrapperView: IvOutcomeWrapperView)
 
-@(allowContinue: Boolean)(implicit request: Request[_], configDecorator: ConfigDecorator, partialRetriever: LocalPartialRetriever, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages)
+@(allowContinue: Boolean)(implicit request: Request[_], configDecorator: ConfigDecorator, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages, ec: ExecutionContext)
 
-@views.html.iv.ivOutcomeWrapper(messages("label.we_cannot_confirm_your_identity"), messages("label.we_cannot_confirm_your_identity")) {
+@ivOutcomeWrapperView(messages("label.we_cannot_confirm_your_identity"), messages("label.we_cannot_confirm_your_identity")) {
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/iv/failure/LockedOutView.scala.html:26]: Done

File: app/views/iv/failure/FailedIvIncompleteView.scala.html [hunk 23]
```diff
@@ -16,13 +16,15 @@
...
 @import config.ConfigDecorator
 @import uk.gov.hmrc.renderer.TemplateRenderer
-@import util.LocalPartialRetriever
 
-@this()
+@import views.html.iv.IvOutcomeWrapperView
+@import scala.concurrent.ExecutionContext
 
-@(retryUrl: String)(implicit request: Request[_], configDecorator: ConfigDecorator, partialRetriever: LocalPartialRetriever, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages)
+@this(ivOutcomeWrapperView: IvOutcomeWrapperView)
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/iv/failure/FailedIvIncompleteView.scala.html:23]: Done

File: app/views/tags/countryInput.scala.html [hunk 24]
```diff
@@ -42,7 +42,7 @@
         <p>@elements.args.get('_additionalTitleText)</p>
     }
     @if(elements.args.contains('_errorsAboveInput)) {
-        @elements.errors.map { error => <span style="display: block"  id="@elements.args.get('_errorId)" class="error-message">@messages(error)</span> }
+        @elements.errors.map { error => <span style="display: block"  id="@elements.args.get('_errorId)" class="error-message">@messages(error.toString)</span> }
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/tags/countryInput.scala.html:45]: Done

File: app/views/tags/input.scala.html [hunk 25]
```diff
@@ -30,7 +30,7 @@
     <label for="@elements.field.name" class="@if( elements.args.get('_labelClass) ){ @elements.args.get('_labelClass) }" @if(elements.args.contains('_labelDataAttributes)){@elements.args.get('_labelDataAttributes)}>
 
         @if(parentElements.isDefined) {
-            @parentElements.get.errors.map { error => <span class="error-notification">@Messages(error)</span>}
+            @parentElements.get.errors.map { error => <span class="error-notification">@Messages(error.toString)</span>}
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/tags/input.scala.html:33]: Done

File: app/views/tags/input_yes_no.scala.html [hunk 26]
```diff
@@ -31,7 +31,7 @@ <h1 class="heading-xlarge">@elements.args.get('_legend)</h1>
 
         @if(elements.args.contains('_hint)){@elements.args.get('_hint)}
 
-        @elements.errors.map{error => <span class="error-message" id="error-message-@{field.id}-input"><span class="visually-hidden">@messages("error.message.error") </span>@Messages(error)</span>}
+        @elements.errors.map{error => <span class="error-message" id="error-message-@{field.id}-input"><span class="visually-hidden">@messages("error.message.error") </span>@Messages(error.toString)</span>}
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply app/views/tags/input_yes_no.scala.html:34]: Done

File: build.sbt [hunk 27]
```diff
@@ -16,10 +16,10 @@ val appName = "pertax-frontend"
 
 scalaVersion := "2.12.11"
 
-val silencerVersion = "1.6.0"
+val silencerVersion = "1.7.0"
 
 lazy val plugins: Seq[Plugins] =
-  Seq(play.sbt.PlayScala, SbtAutoBuildPlugin, SbtGitVersioning, SbtDistributablesPlugin, SbtArtifactory)
+  Seq(play.sbt.PlayScala, SbtAutoBuildPlugin, SbtDistributablesPlugin, PlayNettyServer)
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply build.sbt:22]: Done

File: build.sbt [hunk 28]
```diff
@@ -85,6 +85,7 @@ lazy val microservice = Project(appName, file("."))
       "uk.gov.hmrc.http.HeaderCarrier"
     )
   )
+  .settings(PlayKeys.devSettings += "play.server.provider" -> "play.core.server.NettyServerProvider")
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply build.sbt:88]: Done

File: conf/application.conf [hunk 29]
```diff
@@ -16,19 +16,30 @@ include "frontend.conf"
...
+
+# Provides an implementation and configures all filters required by a Platform frontend microservice.
+play.modules.enabled += "uk.gov.hmrc.play.bootstrap.frontend.FrontendModule"
 
-play.http.filters = "filters.Filters"
+play.modules.enabled += "play.modules.reactivemongo.ReactiveMongoModule"
+
+play.modules.enabled += "modules.LocalGuiceModule"
+
+play.server.provider = play.core.server.AkkaHttpServerProvider
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply conf/application.conf:42]: Done

File: conf/application.conf [hunk 30]
```diff
@@ -198,7 +209,7 @@ microservice {
 }
 
 auditing {
-  enabled = true
+  enabled = false
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply conf/application.conf:212]: Done

File: test/controllers/controllershelpers/PaperlessInterruptHelperSpec.scala [hunk 31]
```diff
@@ -59,35 +59,35 @@ class PaperlessInterruptHelperSpec extends BaseSpec with MockitoSugar {
...
         when(paperlessInterruptHelper.preferencesFrontendService.getPaperlessPreference()(any())) thenReturn {
           Future.successful(ActivatePaperlessRequiresUserActionResponse("/activate-paperless"))
         }
 
         when(mockConfigDecorator.enforcePaperlessPreferenceEnabled).thenReturn(true)
 
-        val r = paperlessInterruptHelper.enforcePaperlessPreference(Ok)
-        status(r) shouldBe SEE_OTHER
-        redirectLocation(await(r)) shouldBe Some("/activate-paperless")
+        val r = paperlessInterruptHelper.enforcePaperlessPreference(Future(Ok))
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply test/controllers/controllershelpers/PaperlessInterruptHelperSpec.scala:70]: Done

File: test/controllers/controllershelpers/PaperlessInterruptHelperSpec.scala [hunk 32]
```diff
@@ -59,35 +59,35 @@ class PaperlessInterruptHelperSpec extends BaseSpec with MockitoSugar {
...
       }
 
       "Return the result of the block when getPaperlessPreference does not return ActivatePaperlessRequiresUserActionResponse" in {
         when(paperlessInterruptHelper.preferencesFrontendService.getPaperlessPreference()(any())) thenReturn {
           Future.successful(ActivatePaperlessNotAllowedResponse)
         }
 
-        val result = await(paperlessInterruptHelper.enforcePaperlessPreference(okBlock))
-        result shouldBe okBlock
+        val result = paperlessInterruptHelper.enforcePaperlessPreference(Future(okBlock))
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply test/controllers/controllershelpers/PaperlessInterruptHelperSpec.scala:80]: Done

File: test/services/UpdateAddressResponseSpec.scala (hunk 1 above)
Comment (Christian-QA) [reply test/services/UpdateAddressResponseSpec.scala:54]: Done

File: test/services/UpdateAddressResponseSpec.scala (hunk 2 above)
Comment (Christian-QA) [reply test/services/UpdateAddressResponseSpec.scala:59]: Done

File: test/services/UpdateAddressResponseSpec.scala (hunk 3 above)
Comment (Christian-QA) [reply test/services/UpdateAddressResponseSpec.scala:65]: Done

File: test/services/UpdateAddressResponseSpec.scala (hunk 4 above)
Comment (Christian-QA) [reply test/services/UpdateAddressResponseSpec.scala:71]: Done

File: test/services/http/SimpleHttpSpec.scala (hunk 5 above)
Comment (Christian-QA) [reply test/services/http/SimpleHttpSpec.scala:28]: Done

File: test/views/html/HomeViewSpec.scala [hunk 33]
```diff
@@ -30,7 +29,7 @@ import viewmodels.HomeViewModel
 
 import scala.collection.JavaConversions._
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply test/views/html/HomeViewSpec.scala:30]: Doing later

File: project/AppDependencies.scala [hunk 34]
```diff
@@ -1,35 +1,35 @@
...
-    "com.github.tomakehurst"   % "wiremock-standalone"  % "2.17.0",
-    "com.typesafe.play"       %% "play-test"            % PlayVersion.current
+    "org.scalatest"           %% "scalatest"                % "3.2.8",
+    "com.typesafe.play"       %% "play-test"                % current,
+    "org.scalatestplus.play"  %% "scalatestplus-play"       % "4.0.3",
+    "org.scalatestplus"       %% "mockito-3-4"              % "3.2.3.0",
+    "org.mockito"             %  "mockito-core"             % "3.6.28",
+    "org.scalacheck"          %% "scalacheck"               % "1.15.1",
+    "com.github.tomakehurst"  %  "wiremock-standalone"      % "2.27.2",
+    "com.vladsch.flexmark"    % "flexmark-all"              % "0.36.8"
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply project/AppDependencies.scala:32]: Done

File: test/config/ConfigDecoratorSpec.scala (hunk 7 above)
Comment (Christian-QA) [reply test/config/ConfigDecoratorSpec.scala:20]: Done

File: test/util/MessagesSpec.scala (hunk 8 above)
Comment (Christian-QA) [reply test/util/MessagesSpec.scala:25]: Done

File: test/views/html/integration/mainContentHeaderSpec.scala (hunk 9 above)
Comment (Christian-QA) [reply test/views/html/integration/mainContentHeaderSpec.scala:23]: Done

File: app/views/UnauthenticatedMainView.scala.html (hunk 10 above)
Comment (Christian-QA) [reply app/views/UnauthenticatedMainView.scala.html:14]: Done

File: app/services/UpdateAddressResponse.scala (hunk 14 above)
Comment (Christian-QA) [reply app/services/UpdateAddressResponse.scala:34]: Done

File: test/controllers/auth/SelfAssessmentStatusActionSpec.scala [hunk 35]
```diff
@@ -38,7 +40,7 @@ import uk.gov.hmrc.domain.{Nino, SaUtr, SaUtrGenerator}
 import scala.concurrent.Future
 
 class SelfAssessmentStatusActionSpec
-    extends FreeSpec with MustMatchers with MockitoSugar with BeforeAndAfterEach with GuiceOneAppPerSuite {
+    extends AnyFreeSpec with Matchers with MockitoSugar with BeforeAndAfterEach with GuiceOneAppPerSuite {
```
Comment (brendanstraughan) [inline test/controllers/auth/SelfAssessmentStatusActionSpec.scala:43]: All apart from AnyFreeSpec will come from BaseSpec

File: test/controllers/auth/requests/UserRequestSpec.scala [hunk 36]
```diff
@@ -16,13 +16,14 @@
...
-import org.scalatestplus.play.PlaySpec
+import models._
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
 import play.api.test.FakeRequest
 import uk.gov.hmrc.domain.SaUtr
+import util.UserRequestFixture._
 
-class UserRequestSpec extends PlaySpec {
+class UserRequestSpec extends AnyWordSpec with Matchers {
```
Comment (brendanstraughan) [inline test/controllers/auth/requests/UserRequestSpec.scala:26]: Could just use BaseSpec

File: test/filters/SessionIdFilterSpec.scala [hunk 37]
```diff
@@ -45,7 +45,7 @@ object SessionIdFilterSpec {
       extends SessionIdFilter(mat, UUID.fromString(sessionId), ec)
 }
 
-class SessionIdFilterSpec extends WordSpec with MustMatchers with OneAppPerSuiteWithComponents {
+class SessionIdFilterSpec extends AnyWordSpec with Matchers with OneAppPerSuiteWithComponents {
```
Comment (brendanstraughan) [inline test/filters/SessionIdFilterSpec.scala:48]: Could just use BaseSpec

File: test/services/NinoDisplayServiceSpec.scala [hunk 38]
```diff
@@ -36,7 +37,7 @@ import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Random
 
 class NinoDisplayServiceSpec
-    extends FreeSpec with MustMatchers with MockitoSugar with ScalaFutures with GuiceOneAppPerSuite {
+    extends AnyFreeSpec with Matchers with MockitoSugar with ScalaFutures with GuiceOneAppPerSuite {
```
Comment (brendanstraughan) [inline test/services/NinoDisplayServiceSpec.scala:40]: All apart from AnyFreeSpec come from BaseSpec

Comment (Christian-QA) [reply test/services/NinoDisplayServiceSpec.scala:40]: Done

File: test/controllers/auth/requests/UserRequestSpec.scala (hunk 36 above)
Comment (Christian-QA) [reply test/controllers/auth/requests/UserRequestSpec.scala:26]: Done

File: test/filters/SessionIdFilterSpec.scala (hunk 37 above)
Comment (Christian-QA) [reply test/filters/SessionIdFilterSpec.scala:48]: Done

File: test/controllers/auth/SelfAssessmentStatusActionSpec.scala (hunk 35 above)
Comment (Christian-QA) [reply test/controllers/auth/SelfAssessmentStatusActionSpec.scala:43]: Done

File: test/controllers/InterstitialControllerSpec.scala [hunk 39]
```diff
@@ -72,7 +71,7 @@ class InterstitialControllerSpec extends BaseSpec with MockitoSugar {
         injected[ViewChildBenefitsSummaryInterstitialView],
         injected[SelfAssessmentSummaryView],
         injected[Sa302InterruptView]
-      )(mock[LocalPartialRetriever], injected[ConfigDecorator], injected[TemplateRenderer], injected[ExecutionContext]) {
+      )(injected[ConfigDecorator], injected[TemplateRenderer], injected[ExecutionContext]) {
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply test/controllers/InterstitialControllerSpec.scala:74]: Done

**Review by struong — APPROVED** [review]

File: project/AppDependencies.scala [hunk 40]
```diff
@@ -1,35 +1,35 @@
...
   val test = Seq(
-    "uk.gov.hmrc"             %% "hmrctest"             % "3.9.0-play-26",
-    "org.scalatest"           %% "scalatest"            % "3.0.8",
-    "org.mockito"              % "mockito-all"          % "2.0.2-beta",
-    "org.scalatestplus.play"  %% "scalatestplus-play"   % "3.1.3",
-    "org.pegdown"              % "pegdown"              % "1.6.0",
-    "org.jsoup"                % "jsoup"                % "1.11.3",
-    "com.github.tomakehurst"   % "wiremock-standalone"  % "2.17.0",
-    "com.typesafe.play"       %% "play-test"            % PlayVersion.current
+    "org.scalatest"           %% "scalatest"                % "3.2.8",
```
Comment (struong) [inline project/AppDependencies.scala:25]: @Christian-QA don't think scalatest is supposed to run with scalatestplus-play > v3.0.x we should downgrade this back to 3.0.8

Comment (Christian-QA) [reply project/AppDependencies.scala:25]: Done. Had to redo all the must matchers imports, so the change will look deceptively big

Comment (Christian-QA) [reply project/AppDependencies.scala:25]: I originally needed to upgrade it to 3.2.x because it was throwing
`java.lang.NoSuchMethodError: org.scalactic.BooleanMacro.<init>(Lscala/reflect/macros/whitebox/Context;Ljava/lang/String;)V`

The error doesn't specify a spec, and with the upgrade resolving the issue, I never discovered what caused it. Will give it another look
